common/djangoapps/track/backends/oli.py

```diff
@@ -70,62 +69,66 @@ def send(self, event):
         if not event_data:
+            LOG.info('event_data attribute missing from event for OLI service')
             return None
 
         problem_id = event_data.get('problem_id')
-        if not problem_id:
+        if problem_id is None:
+            LOG.info('problem_id attribute missing from event for OLI service')
+            return None
+
+        grade = event_data.get('grade')
+        if grade is None:
+            LOG.info('grade attribute missing from event for OLI service')
             return None
 
-        success = event_data.get('success')
-        if not success:
+        max_grade = event_data.get('max_grade')
+        if max_grade is None:
+            LOG.info('max_grade attribute missing from event for OLI service')
             return None
 
-        is_correct = success == 'correct'
+        timestamp = event.get('time')
+        if timestamp is None:
+            LOG.info('time attribute missing from event for OLI service')
+            return None
 
         # put the most expensive operation (DB access) at the end, to not do it needlessly
         try:
             user = User.objects.get(pk=user_id)
         except User.DoesNotExist:
-            LOG.warning('Can not find a user with user_id: %s', user_id)
+            LOG.info('Can not find a user with user_id: %s', user_id)
             return None
 
-        payload = {
-            'course_id': course_id,
-            'resource_id': problem_id,
-            'student_id': anonymous_id_for_user(user, None),
-            'result': is_correct,
-        }
+        request_payload_string = json.dumps({
+            'payload': {
+                'course_id': course_id,
+                'resource_id': problem_id,
+                'student_id': anonymous_id_for_user(user, None),
+                'grade': grade,
+                'max_grade': max_grade,
+                'timestamp': timestamp.isoformat(),
+            }
+        })
+
 
-        request_payload_string = json.dumps({'payload': json.dumps(payload)})
-        request_payload = {'request': request_payload_string}
         endpoint = urljoin(self.url, self.path)
         try:
-            response = self.oauth.put(endpoint, request_payload)
-
-            # Note: CourseBuilder's API always returns status_code=200, regardless of the actual error,
-            # so response.status_code is misleading.
-            # You should check the actual contents of the response.
-            # The response body will look like {"status":200, "message":"OK", "payload": .....}.
-
-            # Because Google prepends ')]}'\n to all json responses, to protect
-            # against XSSI (cross-site-scripting-inclusion), need to reformat the
-            # returned payloads
-
-            message = json.loads(response.content.split('\n')[-1])
-            if message['status'] == 200:
+            response = self.oauth.put(endpoint, request_payload_string)
+            if response.status_code == 200:
```

Comment:
Only the call to `oauth.put` can generate an exception here, so that should be the only line in the `try` block; the `if/else` can be placed after the `except`.